node/heartbeat/monitor.go

```diff
@@ -191,6 +271,8 @@ func (m *Monitor) GetHeartbeats() []PubKeyHeartbeat {
 	m.mutHeartbeatMessages.RLock()
```

Comment:
I do not think RLock can be still used here as we call updateAllHeartbeatMessages that change data inside the structs. Since multiple RLocks are permitted to be ran in parallel, there is a chance that will get unpredictable results.

Reply:
yes! set to Lock and Unlock